### PR TITLE
[dev] checks if port is available before attempting to start server

### LINKED ISSUE #1267: [dev] panic when already running wrangler dev in another terminal
### 🐛 Bug Report

A search of issues reveals this is briefly mentioned in someone's feedback in #1047 , but I figured it's worth having its own issue.

#### Environment

* operating system: macOS 10.14.6 (18G103)
* output of `rustc -V`: `rustc 1.41.0 (5e1a79984 2020-01-27)`
* output of `node -v`: `v12.16.2`
* output of `wrangler -V`: `👷 ✨  wrangler 1.9.0`
* contents of `wrangler.toml`: 
```
name = "typescript-demo"
type = "webpack"
account_id = "e733c486ff936cfb7cdf644badc953e2"
workers_dev = true
route = ""
zone_id = ""
webpack_config = "webpack.config.js"
```

#### Steps to reproduce

1. Open a terminal and navigate to the root of a `wrangler` project.
1. Run `wrangler dev`
1. Open another terminal and navigate to the same directory as previously.
1. Run `wrangler dev`

#### What did you expect to see?

A polite error message and a graceful shutdown.

#### What did you see instead?

```
➜ wrangler dev

[ truncated ]

⬇️ Installing wranglerjs...
⬇️ Installing wasm-pack...
Warning: webpack's output filename is being renamed to worker.js because of requirements from the Workers runtime
✨  Built successfully, built project size is 5 KiB.
thread 'tokio-runtime-worker' panicked at 'error binding to 127.0.0.1:8787: error creating server listener: Address already in use (os error 48)', /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.13.5/src/server/mod.rs:124:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
⬇️ Installing wranglerjs...
Error: panic
```

### PR DESCRIPTION
**Pull request by EverlastingBugstopper**
fixes #1267 

checks if ip/port combo is in use and returns a nicer error for the user if it is :)

### CODE REVIEW

File: src/commands/dev/server_config/mod.rs [hunk 1]
```diff
@@ -14,9 +18,11 @@ impl ServerConfig {
     pub fn new(
         host: Option<&str>,
         ip: Option<&str>,
-        port: Option<&str>,
+        port: Option<u16>,
     ) -> Result<Self, failure::Error> {
-        let port = port.unwrap_or("8787");
+        println!("finding open port from ${:?}", port);
+        let port = find_open_port(port, DEFAULT_PORT)?;
+        println!("found open port: {:?}", port);
```
Comment (EverlastingBugstopper) [inline src/commands/dev/server_config/mod.rs:25]: if i start a dev process w/using default 8787 port, and then start another one, this prints 

`finding open port from None`
`found open port: 8787`

and then it tries to bind to 8787 and it panics w/address already in use

File: src/http/mod.rs [hunk 2]
```diff
@@ -1,7 +1,9 @@
 pub(self) mod cf;
 pub(crate) mod feature;
 pub(self) mod legacy;
+mod port;
```
Comment (EverlastingBugstopper) [inline src/http/mod.rs:4]: do we want this somewhere that is not couple up with our `http` module stuff? i literally would put this in utils lol. could also just make top level port.rs but that feels messy

**Comment by EverlastingBugstopper** [comment]
@ashleymichal when you get the chance could I get a review?

File: src/main.rs [hunk 3]
```diff
@@ -659,7 +659,9 @@ fn run() -> Result<(), failure::Error> {
         commands::preview(target, user, method, url, body, watch, verbose, headless)?;
     } else if let Some(matches) = matches.subcommand_matches("dev") {
         log::info!("Starting dev server");
-        let port = matches.value_of("port");
+        let port: Option<u16> = matches
```
Comment (ashleymichal) [inline src/main.rs:662]: oh snap i need to do this for tail...

**Pull request merged**